Add v0 guards (#50)
Concordium/concordium-node-sdk-js: README.md

const nationality: string = accountInfo.accountCredentials[0].value.contents.policy.revealedAttributes["nationality"];
```

To check if the account is a baker or a delegator, one can use the functions `isDelegatorAccount` and `isBakerAccount`.
```js
...
const accountInfo: AccountInfo = await client.getAccountInfo(accountAddress, blockHash);
if (isDelegatorAccount(accountInfo)) {
    const delegationDetails = accountInfo.accountDelegation;
    ...
} else if (isBakerAccount(accountInfo) {
    const bakingDetails = accountInfo.accountBaker;
    ...
} else {
    // Neither a baker nor a delegator
}
```
Furthermore there are different versions, based on Protocol version, of a baker's accountInfo.

> In protocol version 4 the concept of baker pools were introduced, so to get baker pool information, one should confirm the version first with `isBakerAccountV0` or `isBakerAccountV1`.

Reviewer: ```suggestion
In protocol version 4 the concept of baker pools was introduced, so to get baker pool information one should confirm the version with `isBakerAccountV0` or `isBakerAccountV1`.
```

```js
...
const accountInfo: AccountInfo = await client.getAccountInfo(accountAddress, blockHash);
if (isBakerAccountV1(accountInfo)) {
    const bakerPoolInfo = accountInfo.accountBaker.bakerPoolInfo;

>     ...

Reviewer: Add example of what to do when using `isBakerAccountV0`.

...
```

Blocks before protocol version 4 have a different type than those from higher protocol versions.
To determine the version, use `isBlockSummaryV1` and `isBlockSummaryV0`:

```js
...
const blockSummary: BlockSummary = await client.getBlockSummary(blockHash);
if (isBlockSummaryV0(blockSummary)) {

>     // blockSummary has version 0 structure 

Reviewer: Maybe we should mention that this corresponds to protocol version <= 3, and likewise for version 1.

...
```

## getRewardStatus
Retrieves the current amount of funds in the system at a specific block, and the state of the special accounts. 
Protocol version 4 expanded the amount of information in the response, so one should check the type to access that.
This information includes information about the payday and total amount of funds staked.
```js
const blockHash = "7f7409679e53875567e2ae812c9fcefe90ced8961d08554756f42bf268a42749";

const rewardStatus = await client.getRewardStatus(blockHash);

> if (isRewardStatusV1()) {

Reviewer: An example with `isRewardStatusV0()`.

```
Finally, to actually update the contract on the chain, send the constructed `updateContractTransaction` to the chain using `sendAccountTransaction`. (See [Send Account Transaction](#Send-Account-Transaction) for how to do this)

## getInstances
Used to get the full list of contract instances on the chain at a specific block.
```js
const blockHash = "7f7409679e53875567e2ae812c9fcefe90ced8961d08554756f42bf268a42749";

const instances = await client.getInstances(blockHash);
...
```

## getInstanceInfo
Used to get information about a specific contract instance, at a specific block.

```js
const blockHash = "7f7409679e53875567e2ae812c9fcefe90ced8961d08554756f42bf268a42749";
const contractAddress = { index: 1n, subindex: 0n };

> const instanceInfo = await client.getPoolStatus(blockHash, contractAddress);

Reviewer: I don't think `getPoolStatus` does this.
